src/__tests__/index.test.js

```diff
@@ -10,19 +11,54 @@ beforeEach(() => {
     })
 })
 
-test('processEvent filters event', async () => {
+test('processEvent filters event', () => {
+    // Setup Plugin
+    setupPlugin(getMeta())
+
+    for (let i = 0; i < 100; i++) {
+        const event0 = createEvent({ distinct_id: randomBytes(10).toString('hex') })
+        const event1 = processEvent(event0, getMeta())
+        expect(event1).toEqual(event0)
+    }
+})
+
+test('processEvent filters 0 events at 0 percent', () => {
+    resetMeta({
+        config: {
+            percentage: '0',
+        },
+    })
+
+    // Setup Plugin
+    setupPlugin(getMeta())
+
     // create a random event
     const event0 = createEvent({ event: 'blah' })
 
-    // TODO: Figure out how getMeta can manipulate meta
-
-    for(let i=0; i < 100; i++) {
-        const event1 = await processEvent(event0, getMeta())
-        expect(event1).toEqual(event0)
+    for (let i = 0; i < 100; i++) {
+        const event1 = processEvent(event0, getMeta())
+        expect(event1).toBeNull()
     }
 })
 
 
-test('processEvent filters 0 events at 0 percent', async () => {
-    // TODO
+
+test('processEvent filters same events at different increasing percent', () => {
+
+    // create a event. Hash generates 0.42
+    const event0 = createEvent({ distinct_id: '1' })
+
+    for (let i = 5; i < 10; i++) {
```

Comment:
Check also the failure case, that the event isn't there for `i = 0 .. 4`

Reply:
Nice, good idea